controllers/teamdetail.controller.js

```
@@ -15,23 +15,86 @@ function getJiraTeamDetail(req, res, next) {
 
     utils.getJiraBackendUrl(clientId)
         .then(jiraUrl => {
-            return axios.get(`${jiraUrl}/jira/velocity`, {
-                params: {
-                    team: team,
-                    startDate: startDate,
-                    endDate: endDate
-                }
-            });
+        	return getJiraTeamStats(jiraUrl, team, startDate, endDate);
         })
-        .then(response => {
+        .then(responsesData => {
             res.json({
-                velocities: response.data
+                velocities: responsesData[0],
+                focuses: responsesData[1]
             });
         }).catch(err => {
             next(err);
         })
 }
 
+function getJiraTeamStats(jiraUrl, team, startDate, endDate) {
+	const teamStatsPromises = [];
+	teamStatsPromises.push(getTeamVelocity(jiraUrl, team, startDate, endDate));
+	teamStatsPromises.push(getTeamFocus(jiraUrl, team, startDate, endDate));
+	return Promise.all(teamStatsPromises);
+}
+
+function getTeamVelocity(jiraUrl, team, startDate, endDate) {
+	return axios.get(`${jiraUrl}/jira/velocity`, {
+        params: {
+            team: team,
+            startDate: startDate,
+            endDate: endDate
+        }
+    })
+    .then(response => {
+        return response.data;
+    })
+    .catch(error => {
+    	if(error.response) {
+    		console.log(error.response.status);
+    		console.log(error.response.data);
+    		return {
+                status: error.response.status,
+                message: error.response.data
+    		}
+    	}
+    	else {
+			console.log(error);
+    		return {
+                status: 500,
+                message: "Couln't retrieve team Velocity " + error.message
```

If I see correctly, this only returns JSON error object for a specific metric - in this case veocity, but it will still return data for focus factor right? This doesn't mean that the whole request will end in Http 500 response right?

---

Yes, I hope that's right (?)

---

Not sure what best practices are in this case but for now let's return this error element.

---

I guess, it will look like this when e.g. focus request fails:
velocities: {….}
focuses: { status: 500, message: '....'}
It needs to be documented in middleware-spec.yaml